src/Radicle/Ipfs.hs

```diff
@@ -188,10 +189,21 @@ instance FromJSON DagPutResponse where
             Left _    -> fail "invalid CID"
             Right cid -> pure $ DagPutResponse cid
 
+newtype PinResponse = PinResponse [CID]
 
+instance FromJSON PinResponse where
+  parseJSON = Aeson.withObject "v0/pin/add response" $ \o -> do
+    cidTexts <- o .: "Pins"
+    case traverse cidFromText cidTexts of
+      Left _     -> fail "invalid CID"
+      Right cids -> pure $ PinResponse cids
+
+-- | Get and pin a DAG node.
 dagGet :: FromJSON a => Address -> IO a
 dagGet addr = do
-    result <- ipfsHttpGet "dag/get" [("arg", addressToText addr)]
+    let params = [("arg", addressToText addr)]
+    result <- ipfsHttpGet "dag/get" params
+    PinResponse _ <- ipfsHttpGet "pin/add" params
```

Comment:
I would prefer to introduce a seperate `pinAdd` method. The `Radicle.Ipfs` module should mirror the IPFS API. It is then up to the application logic to pin a machine explicitly